### PR TITLE
[FLINK-13373][docs] Remove Tutorials and Examples sections from Getting Started

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change

The new code walkthroughs and docker playgrounds supersede the tutorial and example sections so these sections should be removed or the content moved to a better location. 

This change is much smaller than the diff makes it appear. Most of the files touched are updated links and redirects. 

## Brief changelog

25062ea Remove examples section from getting started and inline `DataSet` examples into the dataset section

74e60e6 The python tutorial is the only section under api tutorials, this commit moves it to code walkthroughs and removes that section

4397bfa Cleanup of grammar in the python tutorial. I did not change any content. 

31ecc45 Moved the local cluster setup to deployment and operations since the docker environment is a better dev-ops intro. `/ops/deployment/local.md` is the only file that needs a review, every other change in this commit is updating links. **Important:** This page renders differently depending on whether the config is set to snapshot or stable. 

## Verifying this change

I ran ./check_links.sh to verify no broken links.

### CODE REVIEW

**Comment by sjwiesman** [comment]
@knaufk Would you like to take a look at this?

File: docs/ops/deployment/local.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+</div>
+
+{% else %}
+### Download and Compile
+Clone the source code from one of our [repositories](https://flink.apache.org/community.html#source-code), e.g.:
+
+{% highlight bash %}
+$ git clone https://github.com/apache/flink.git
+$ cd flink
+$ mvn clean package -DskipTests # this will take up to 10 minutes
```
Comment (StephanEwen) [inline docs/ops/deployment/local.md]: I think this takes more than 10 minutes by now ;-)

One way to further accelerate is to use `mvn clean package -DskipTests -Pfast`
@zentol

**Comment by StephanEwen** [comment]
Looks pretty good, thanks! One minor comment, otherwise good to merge from my side.

File: docs/ops/deployment/local.md (hunk 1 above)
Comment (zentol) [reply docs/ops/deployment/local.md]: yeah it's more like 25 I think. Depending on the environment the build of the WebUI may also slow things down significantly; or fail outright. May be useful to document `-Pskip-webui-build` for these cases.

**Comment by sjwiesman** [comment]
Thanks for taking a look @StephanEwen, I've pushed an update to address your comment.

File: docs/ops/deployment/local.md (hunk 1 above)
Comment (StephanEwen) [reply docs/ops/deployment/local.md]: Fair point. WSL cannot build the web-ui at the moment, simply deadlocks at some point :-(

**Comment by sjwiesman** [comment]
@rmetzger would you be able to merge this?

**Comment by rmetzger** [comment]
Thanks!

**Pull request merged**